vulnerabilitylookupsighting/publish.py

```diff
@@ -21,61 +19,37 @@ def remove_case_insensitive_duplicates(input_list: list[str]) -> list[str]:
 def push_sighting_to_vulnerability_lookup(attribute, vulnerability_ids):
     """Create a sighting from a MISP attribute and push it to the Vulnerability Lookup instance."""
     print("Pushing sightings to Vulnerability Lookup...")
-    headers_json = {
-        "Content-Type": "application/json",
-        "accept": "application/json",
-        "X-API-KEY": f"{config.vulnerability_auth_token}",
-    }
+    vuln_lookup = PyVulnerabilityLookup(config.vulnerability_lookup_base_url, token=config.vulnerability_auth_token)
     for vuln in vulnerability_ids:
         # Set the creation timestamp for the sighting
         creation_timestamp = ""
-        if attribute.get("first_seen", False):
+        if hasattr(attribute, "first_seen") and attribute.first_seen:
             try:
-                creation_timestamp = datetime.fromtimestamp(
-                    int(attribute["first_seen"]), timezone.utc
-                ).isoformat()
+                creation_timestamp = attribute.first_seen.isoformat()
             except Exception:
-                print(
-                    "Bad timestamp: {} for event {}".format(
-                        attribute["first_seen"], attribute["Event"]["uuid"]
-                    )
-                )
+                print(f"Bad timestamp: {attribute.first_seen} for event {attribute.uuid}")
                 continue
         else:
             try:
-                creation_timestamp = datetime.fromtimestamp(
-                    int(attribute["timestamp"]), timezone.utc
-                ).isoformat()
+                creation_timestamp = attribute.timestamp.isoformat()
             except Exception:
-                print(
-                    "Bad timestamp: {} for event {}".format(
-                        attribute["timestamp"], attribute["Event"]["uuid"]
-                    )
-                )
+                print(f"Bad timestamp: {attribute.timestamp} for event {attribute.uuid}")
                 continue
         if not creation_timestamp:
             continue
 
         # Create the sighting
         sighting = {
             "type": "seen",
-            "source": f"MISP/{attribute['Event']['uuid']}",
+            "source": f"MISP/{attribute.uuid}",
             "vulnerability": vuln,
             "creation_timestamp": creation_timestamp,
```

Comment:
as I said in [here](https://github.com/cve-search/PyVulnerabilityLookup/issues/67#issuecomment-2484199633), current_timestamp is a string at this point. Which I think is good. But later in the PyVulnerabilityLookup it will crash on .tzinfo.

Reply:
I'll patch PyVulnerabilityLookup so it accepts a string, but PyMISP gives a datetime, so we can just use that (the timestamp has a tzinfo set, I'm checking the status of first_seen)

Reply:
confirmed, first seen from MISP has a TZ info set too.

Reply:
ok, thank you ! No hurry.  I need to sleep now ;-)

Reply:
gn! :zzz: